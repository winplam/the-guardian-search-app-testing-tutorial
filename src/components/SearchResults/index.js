import React from 'react'
import PropTypes from 'prop-types'

const SearchResults = ({ articles = [] }) => (
  <ul>
    {articles.map(({ webUrl, webTitle }) => (
      <li key={webUrl}>
        <a href={webUrl} target="_blank" rel="noopener noreferrer">
          {webTitle}
        </a>
      </li>
    ))}
  </ul>
)

SearchResults.propTypes = {
  articles: PropTypes.array
}

export default SearchResults
