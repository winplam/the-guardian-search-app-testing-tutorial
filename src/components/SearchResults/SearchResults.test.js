import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import SearchResults from './'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })

const mockArticles = [
  {
    webUrl: 'http://google.com',
    webTitle: 'This is Google'
  }
]

describe('SearchResults component', () => {
  test('renders', () => {
    const wrapper = shallow(<SearchResults articles={mockArticles}/>)
    expect(wrapper).toMatchSnapshot()
  })

  test('returns the default empty array when there is no data to map through' +
    '\n or what would happen when I render the component with no data', () => {
    const wrapper = shallow(<SearchResults/>)
    expect(wrapper).toMatchSnapshot()
  })

  test('doesn\'t break without articles', () => {
    const wrapper = shallow(<SearchResults/>)
    expect(wrapper.find('li')).toHaveLength(0)
  })

  test('doesn\'t break with an empty array', () => {
    const wrapper = shallow(<SearchResults articles={[]}/>)
    expect(wrapper.find('li')).toHaveLength(0)
  })
})
