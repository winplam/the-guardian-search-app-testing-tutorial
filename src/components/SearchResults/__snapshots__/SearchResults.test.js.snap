// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults component renders 1`] = `ShallowWrapper {}`;

exports[`SearchResults component returns the default empty array when there is no data to map through
 or what would happen when I render the component with no data 1`] = `ShallowWrapper {}`;
